src/store/modules/TsModsModule.ts

```diff
@@ -277,6 +289,8 @@ export const TsModsModule = {
             // hash is updated in the API.
             if (successes === chunkCount) {
                 await dispatch('cacheIndexHash', packageListIndex.hash);
+            } else {
+                throw new Error('Failed to fetch and cache some of the packages.');
```

Comment:
- If I'm reading this right. throwing here turns the partial failure to complete failure since `syncPackageList` skips the rest of the update. Instead, I would try to throw (or just commit) the error message at the end of `syncPackageList`'s try-block if all chunks weren't successful
- Not using `PARTIAL_UPDATE_ERROR` makes me seriously question how much effort you put into this PR